scipy/stats/_distn_infrastructure.py

```diff
@@ -554,18 +555,20 @@ def _parse_args_stats(self, %(shape_arg_str)s %(locscale_in)s, moments='mv'):
 # The function name ncx2 is an abbreviation for noncentral chi squared.
 
 def _ncx2_log_pdf(x, df, nc):
+    pdf = _ncx2_pdf(x, df, nc)
+    return _lazywhere(pdf > 0, (pdf,), f=np.log, fillvalue=-np.inf)
```

Comment:
`_ncx2_log_pdf` is used needed for [`ncx2._logpdf`](https://github.com/scipy/scipy/blob/1bce83f54924be8844b1b0ad67e528607eaebd61/scipy/stats/_continuous_distns.py#L5764) which is still useful(?) because it allows for calling `chi2.logpdf` more directly in the `nc==0` case.

I suspect right now in the `nc==0` case
`ncx2.logpdf(x, df, 0)` => `chp.logpdf(x, df))`

If I delete this and `ncx2._log_pdf` I think what will happen is
`ncx2.logpdf(x, df, 0)` => `np.log(ncx2.pdf(x, df, 0))` => `np.log(chi.pdf(x, df))` =>  `np.log(np.exp(chp.logpdf(x, df)))`

Opinions?